cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedExpr.qll

```diff
@@ -1275,6 +1277,12 @@ class TranslatedUnaryExpr extends TranslatedSingleInstructionExpr {
     expr instanceof UnaryPlusExpr and result instanceof Opcode::CopyValue
     or
     expr instanceof UnaryMinusExpr and result instanceof Opcode::Negate
+    or
+    // TODO: Use a new opcode to represent "awaiting the value"
+    expr instanceof CoAwaitExpr and result instanceof Opcode::CopyValue
+    or
+    // TODO: Use a new opcode to represent "awaiting the value"
```

Comment:
Shouldn't this be:
```suggestion
    // TODO: Use a new opcode to represent "yielding a value"
```
?

Reply:
Actually no. `co_yield` is just syntactic sugar for:
```cpp
co_await promise.yield_value(expr)
```
(see [here](https://en.cppreference.com/w/cpp/language/coroutines)) and the "yielding" is done by the call to `yield_value`.

So whatever instruction we come up with to use the above TODO should also be used here since a `co_yield` ultimately does something equivalent to a special kind of `co_await`.